sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/ImgWithoutAltCheck.java

```diff
@@ -26,7 +27,8 @@
 
   @Override
   public void startElement(TagNode node) {
-    if ((isImgTag(node) || isImageInput(node)) && !node.hasProperty("ALT")) {
+    if ((isImgTag(node) && !node.hasProperty("ALT")) || 
+        ((isImageInput(node) || isAreaTag(node)) && (!node.hasProperty("ALT") || StringUtils.trim(node.getPropertyValue("ALT")).isEmpty()))) {
```

Comment:
The `if-condition` is now a bit hard to read, these conditions could broken down and refactored into helper functions to improve readability.